In [1]:
import numpy as np
import xarray as xr
import netCDF4
import os
import datetime
import matplotlib.pyplot as plt 
from matplotlib import cm
import shutil
from datetime import datetime, timedelta
import glob
import datetime as dt
from os import path
import os
import fsspec
import git 
import json
import sys
import yaml
import requests
from datetime import timezone

Date: January 2024

Author: Martin Wegmann

Contact: martinwegmann@pm.me


In [2]:
startyear=2020
today= datetime.today()
endyear=2023

## get alplakes 

In [3]:
###### alplakes

In [4]:
startyear=startyear
endyear=endyear
#output_folder=output_folder
lon="6.67"
lat="46.5"
lake="geneva"
csv_save=True
nc_save=True

* I cut the requests into monthly chunks in order to not overwhelm the system. This is still a relict of the meteolakes times and could possibly be abolished in the future. But so far, it does not hurt. 

In [5]:
for year in range(startyear,endyear+1):
    print("Downloading year "+str(year))
    range_months=range(1,12+1)
    if year == today.year:
        range_months=range(1,today.month-1)
    for month in range_months:
        
        if month==12:
            start_date=[year,month,1,3]
            end_date=[year+1,1,1,0]
        else:
            start_date=[year,month,1,3]
            end_date=[year,month+1,1,0]
        start_time=dt.datetime(start_date[0], start_date[1], start_date[2],start_date[3])
        timestamp_start=start_time.strftime('%Y%m%d%H%M')
        end_time=dt.datetime(end_date[0], end_date[1], end_date[2],end_date[3])
        timestamp_end=end_time.strftime('%Y%m%d%H%M')

        #print(timestamp_start)
        
        url_alplakes="https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/"+lake+"/"+str(timestamp_start)+"/"+str(timestamp_end)+"/"+lat+"/"+lon
        print(url_alplakes)
        with requests.Session() as s:
            download = s.get(url_alplakes)
        print(download)

https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202001010300/202002010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202002010300/202003010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202003010300/202004010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202004010300/202005010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202005010300/202006010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202006010300/202007010000/46.5/6.67
<Response [200]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202007010300/202008010000/46.5/6.67
<Response [502]>
https://alplakes-api.eawag.ch/simulations/depthtime/delft3d-flow/geneva/202008010300/202009010000/46.5/6.67
<Response [200]>


* The bad responses (502) are seemingly random and their appearances change with each execution of the script. Even though in this execution there is no 502 response for 2023, this is just random and can not be considered a rule. 